File: CS/ImageSlider/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace ImageSlider
{
    public partial class Form1 : Form
    {
        int currentImageIndex = 0;
        Image[] images;

        public Form1()
        {
            this.images = CreateImages();
            InitializeComponent();
        }

        Image[] CreateImages()
        {
            return new Image[]{
                CreateImage(Color.Red),
                CreateImage(Color.Green),
                CreateImage(Color.Blue),
            };
        }

        Image CreateImage(Color color)
        {
            Bitmap bitmap = new Bitmap(100, 100);
            for (int i = 0; i < bitmap.Width; i++)
            {
                for (int j = 0; j < bitmap.Height; j++)
                {
                    bitmap.SetPixel(i, j, color);
                }
            }
            return bitmap;
        }

        private void imageSlider1_CanGetNextPrevImage(object sender, DevExpress.XtraEditors.Controls.CanGetNextPrevImageEventArgs e)
        {
            if (e.IsNext) e.CanGetImage = (currentImageIndex < images.Length - 1);
            else e.CanGetImage = (currentImageIndex > 0);
        }

        private void imageSlider1_GetImage(object sender, DevExpress.XtraEditors.Controls.GetImageEventArgs e)
        {
            if (e.IsStartUp)
            {
                e.Image = images[0];
                return;
            }
            if (e.IsNext)
            {
                e.Image = images[currentImageIndex + 1];
                if (e.CurrentImageUpdated) ++currentImageIndex;
            }
            else if (e.IsPrev)
            {
                e.Image = images[currentImageIndex - 1];
                if (e.CurrentImageUpdated) --currentImageIndex;
            }
        }
    }
}

File: VB/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms

Namespace ImageSlider
	Public Partial Class Form1
		Inherits Form
		Private currentImageIndex As Integer = 0
		Private images As Image()

		Public Sub New()
			Me.images = CreateImages()
			InitializeComponent()
		End Sub

		Private Function CreateImages() As Image()
			Return New Image(){ CreateImage(Color.Red), CreateImage(Color.Green), CreateImage(Color.Blue) }
		End Function

		Private Function CreateImage(ByVal color As Color) As Image
			Dim bitmap As Bitmap = New Bitmap(100, 100)
			Dim i As Integer = 0
			Do While i < bitmap.Width
				Dim j As Integer = 0
				Do While j < bitmap.Height
					bitmap.SetPixel(i, j, color)
					j += 1
				Loop
				i += 1
			Loop
			Return bitmap
		End Function

		Private Sub imageSlider1_CanGetNextPrevImage(ByVal sender As Object, ByVal e As DevExpress.XtraEditors.Controls.CanGetNextPrevImageEventArgs) Handles imageSlider1.CanGetNextPrevImage
			If e.IsNext Then
			e.CanGetImage = (currentImageIndex < images.Length - 1)
			Else
				e.CanGetImage = (currentImageIndex > 0)
			End If
		End Sub

		Private Sub imageSlider1_GetImage(ByVal sender As Object, ByVal e As DevExpress.XtraEditors.Controls.GetImageEventArgs) Handles imageSlider1.GetImage
			If e.IsStartUp Then
				e.Image = images(0)
				Return
			End If
			If e.IsNext Then
				e.Image = images(currentImageIndex + 1)
				If e.CurrentImageUpdated Then
				currentImageIndex += 1
				End If
			ElseIf e.IsPrev Then
				e.Image = images(currentImageIndex - 1)
				If e.CurrentImageUpdated Then
				currentImageIndex -= 1
				End If
			End If
		End Sub
	End Class
End Namespace
